.changeset/nasty-eagles-reply.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/integration': minor
+---
+
+Automatically use the system-assigned managed identity when `system-assigned` is specified as client ID.
```

Comment:
As these are for the same package please combine the changesets. You can just have these on a new line.

Reply:
I chose to do this in a separate commit because it technically is not related to the federated credential flow. If this is merged "rebase & fast-forward" then it could technically have been two separate pull requests. 

If I understand correctly this wont cause a +2 in the minor version of the package at release time right?

Reply:
Done, I also added a YAML diff to highlight the change in a bit more detail